transport/src/main/java/io/netty/channel/SendDirectByteBufPool.java

```diff
@@ -0,0 +1,1189 @@
+        current.refCnt ++;
+        int writerIndex = buffer.writerIndex();
+        // Slice out a buffer and set reader and writerIndex to 0.
+        ByteBuf slice = buffer.slice(writerIndex, size).clear();
+
+        // Increase writerIndex of buffer from which we sliced out.
+        buffer.writerIndex(writerIndex + size);
+        return new PooledSendByteBuf(current, slice);
+    }
+
+    private Preallocation getPreallocation() {
+        Preallocation current = this.current;
+        if (current.refCnt == 0) {
+            current.buffer.clear();
+            return current;
+        }
+
+        return getPreallocation0();
+    }
+
+    private Preallocation getPreallocation0() {
+        PreallocationRef ref = poolHead;
+        if (ref != null) {
+            do {
+                Preallocation p = ref.get();
+                ref = ref.next;
+
+                if (p != null) {
+                    poolHead = ref;
+                    return p;
+                } else {
+                    ref.release();
+                }
+            } while (ref != null);
+
+            poolHead = ref;
+        }
+
+        return new Preallocation(DEFAULT_PREALLOCATION_SIZE);
+    }
+
+    private static final class Preallocation {
+        final ByteBuf buffer;
+        int refCnt;
+
+        Preallocation(int capacity) {
+            buffer = Unpooled.directBuffer(capacity);
+        }
+    }
+
+    private static final class PreallocationRef extends SoftReference<Preallocation> {
+        final PreallocationRef next;
+        final ByteBuf buffer;
+
+        PreallocationRef(Preallocation prealloation, PreallocationRef next) {
+            super(prealloation);
+            buffer = prealloation.buffer;
+            this.next = next;
+        }
+
+        void release() {
+            ReferenceCountUtil.safeRelease(buffer);
+        }
+    }
+
+    private final class PooledSendByteBuf extends ByteBuf {
+
+        private final Preallocation parent;
+        private final ByteBuf buffer;
+        private final Thread thread;
+        private int refCnt = 1; // Non volatile as we enforce the Thread to be the same for all release / retain calls.
+
+        PooledSendByteBuf(Preallocation parent, ByteBuf buffer) {
+            this.parent = parent;
+            this.buffer = buffer;
+            thread = Thread.currentThread();
+        }
+
+        @Override
+        public int capacity() {
+            return buffer.capacity();
+        }
+
+        @Override
+        public ByteBuf capacity(int newCapacity) {
+            buffer.capacity(newCapacity);
+            return this;
+        }
+
+        @Override
+        public int maxCapacity() {
+            return buffer.maxCapacity();
+        }
+
+        @Override
+        public ByteBufAllocator alloc() {
+            return buffer.alloc();
+        }
+
+        @Override
+        @Deprecated
+        public ByteOrder order() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)